File: 2025-1/Basic/SJP/Array/BOJ_13300.cpp
#include <iostream>
using namespace std;

int main(){
    int N, K;
    cin >> N >> K;

    int M[6] = {0};
    int F[6] = {0};

    int a, b;

    for (int i = 0; i < N; i++){
        cin >> a >> b;
        if (a == 0) F[b - 1]++;
        else M[b-1]++;
    }

    int result = 0;

    for (int i = 0; i < 6; i++){
        result += F[i] / K + (F[i] % K != 0);
        result += M[i] / K + (M[i] % K != 0);
    }
    cout << result;
}

File: 2025-1/Basic/SJP/Array/BOJ_13300.py
import sys
input = sys.stdin.readline

N, K = map(int, input().split())

M = [0] * 6
F = [0] * 6

for _ in range(N):
    a, b = map(int, input().split())
    if a == 0:
        F[b-1] += 1
    else:
        M[b-1] += 1

for i in range(6):
    M[i] = M[i] // K + bool(M[i] % K)
    F[i] = F[i] // K + bool(F[i] % K)

result = sum(M) + sum(F)

print(result)
